README.md

```diff
@@ -14,3 +14,47 @@ part of the array each recursive call considers.
 Analyse the time complexity of your implementation and give a $\Theta$ bound for
 its worst-case runtime. Add your answer, including your reasoning, to this
 markdown file.
+
+#### I've deduced that $T(n) \in \Theta(n^{3}\log{n})$ by this reasoning:
+
+The worst case for mergesort is if every subarray in the data set is out of
+order, so shifting and comparisons will need to happen for every element.
+
+##### The steps in this algorithm are:
+1. For loop to check if i is greater than the array length and increment i to the
+   next power of 2. Each iteration of i will be the next power of 2 all the way
+   up to n for the worst case, so this step will run on at most $\log_{2}n$
+
+    2. A nested for loop to check if k + i is greater than the array length and
+       increment k by 2*i to set the bounds of the current sub array. On the worst 
+       case this will run on at most $n$ since n will be divided by a constant factor
+       for each iteration which does not matter for asymptotic analysis.
+
+       3. A while loop that checks if the merge is finished or if the merge does
+          not need to be done. On the worst case this will iterate over the whole
+          array (n)
```

Comment:
But does it do that for each iteration? This part sounds like it's the same for recursive mergesort, where we got a different complexity.